### ISSUE TITLE
json_file Checksum on resource (6a3150) does not match checksum on content (025106)

### ISSUE THREAD

**Issue by mhalligan** [sensu/sensu-chef#332]
Hi, -- I'm running into a problem with the 0.4.0 cookbook. 50% of the time, running chef-client fails with the error pasted below.

```
sensu_client "#{node['hostname']}.mydomain.infra" do
  address           node.ipaddress
  subscriptions    ["all"]
  keepalive        :handlers => ["chef_node","ec2_node","pagerduty"]
end
```

```
   ================================================================================
    Error executing action `create` on resource 'file[/etc/sensu/conf.d/client.json]'
    ================================================================================

    Chef::Exceptions::ChecksumMismatch
    ----------------------------------
    Checksum on resource (6a3150) does not match checksum on content (025106)

    Resource Declaration:
    ---------------------
    # In /var/cache/chef/cookbooks/sensu/providers/json_file.rb

     14:     f = file new_resource.path do
     15:       owner @owner
     16:       group new_resource.group
     17:       mode new_resource.mode
     18:       content Sensu::JSONFile.dump_json(new_resource.content)
     19:       notifies :create, "ruby_block[sensu_service_trigger]", :delayed
     20:     end
     21:

    Compiled Resource:
    ------------------
    # Declared in /var/cache/chef/cookbooks/sensu/providers/json_file.rb:14:in `block in class_from_file'

    file("/etc/sensu/conf.d/client.json") do
      action "create"
      updated true
      retries 0
      retry_delay 2
      default_guard_interpreter :default
      path "/etc/sensu/conf.d/client.json"
      backup 5
      atomic_update true
      declared_type :file
      cookbook_name "sensu-wraooer"
      content "{\n  \"client\": {\n    \"name\": \"sensu1.hostname.fqdn\",\n    \"address\": \"10.1.11.159\",\n    \"subscriptions\": [\n      \"all\"\n    ],\n    \"keepalive\": {\n      \"handlers\": [\n        \"chef_node\",\n        \"ec2_node\",\n        \"pagerduty\"\n      ]\n    }\n  }\n}\n"
      group "sensu"
      mode 416
      owner "root"
      checksum "6a3150e146168dcb13dab1a803238fdfccda3067df41925472699342c7a9c792"
    end
```

**Comment by johntdyer** [comment]
We are getting the exact same thing..

**Comment by johntdyer** [comment]
chef/chef#2689 :/

**Comment by lamont-granquist** [comment]
This was probably chef/chef#3295 which is fixed.

**Comment by cwjohnston** [comment]
Closing as this seems to be resolved upstream circa chef ~12.4
